test/unit/models/Tag.test.js

```diff
@@ -270,25 +270,49 @@ describe('Tag', () => {
         expect(selected.get('name')).to.equal('tags')
       })
     })
+
+    it('preserves a tag from a comment, and removes a tag not in any of the comments', () => {
+      const post = factories.post({description: '<p>#preexisting</p>'})
+      const tag1 = factories.tag({name: 'preexisting'})
+      const tag2 = factories.tag({name: 'commenttag'})
+      const tag3 = factories.tag({name: 'unexpected'})
+      const comment = factories.comment()
+      return Promise.join(post.save(), tag1.save(), tag2.save(), tag3.save(), comment.save())
+      .then(() => Promise.join(
+        post.tags().attach(tag1),
```

Comment:
you can pass multiple objects into `attach` by passing an array as the first argument